### ISSUE TITLE
Can't get rid of stringing

### ISSUE THREAD

**Issue by EkiciLP** [SoftFever/OrcaSlicer#3150]
### OrcaSlicer Version

1.8.1

### OS version

Windows 10

### Printer

Anycubic Vyper with Direct Extruder mod

### How to reproduce

1. Slice some Object
2. Print it
3. See the Strings

### Actual results

I see many Strings and even Blobs which I don't see if I slice with Superslicer

### Expected results

No stringing. I already tried solutions of other posts like using Spiral Z-Hop which significantly improved the strings but they are still there and sometimes my Parts don't fit together because of blobs. I don't know if I just didn't copy the settings correctly but it seems right to me. I've been troubleshooting with different speeds and lengths but the results are always almost identical. With Superslicer there are other issues and in my opinion Orca Slicer is just way better, partly because it is actually maintained.

I will attach both my Orcaslicer and Superslicer profiles so you can check that I haven't done some dumb mistake.

### Project file & Debug log uploads

[Configs.zip](https://github.com/SoftFever/OrcaSlicer/files/13693610/Configs.zip)

### Checklist of files to include

- [ ] Log file
- [ ] Project file

**Comment by oscillatingdecay** [comment]
I am also experiencing this. I think there is something wrong with "retraction" was a whole. I am experiencing the string but if you look at the first layer, the seam is blobbing and not retracting. As soon as I imported the settings and sliced the model in Prusaslicer, the blobs on the first layer went away where the reatraction would have occurred.

**Comment by NeverUsedID** [comment]
The reason is "wipe while retraction" on my side.  I also can't see any retraction g-code in preview or in exported gcode file. So only Wipe without retraction is happening.

Example1: "retraction on layer change" enabled, but no wipe:

![grafik](https://github.com/SoftFever/OrcaSlicer/assets/7545210/fc436278-1984-4765-b85d-5ee70c2b3db5)

Example2: Wipe enabled
![grafik](https://github.com/SoftFever/OrcaSlicer/assets/7545210/0c6bffd9-ba22-4800-9c60-c73dbc564680)

I get the stringing exactly on the wipe positions allong these travel lines:

![grafik](https://github.com/SoftFever/OrcaSlicer/assets/7545210/7ae9a707-3938-49c9-b390-25c8d69b3f66)

Keep in mind that i'am really new to 3d-printing.

**Comment by kadisonm** [comment]
I'm experiencing the same problem and it looks like other people have been as well. 

https://www.reddit.com/r/FixMyPrint/comments/1764ewy/stringing_after_switching_to_orca_slicer/

It seems like this [user](https://www.reddit.com/r/FixMyPrint/comments/1764ewy/comment/k9quany/?utm_source=share&utm_medium=web2x&context=3) commented something similar to @NeverUsedID, so maybe this has something to do with the stringing issues?

I've done a couple of things to prevent stringing but I'm still getting it especially on my organic supports. It isn't horrible but it is way more than what I get on Bambu Studio. If it's relevant I am using a Bambu Lab P1S with Sunlu PLA+.

What I've tried:
- Increasing retraction length (.9mm)
- Spiral Z hop
- Retract on layer change
- Setting retraction speed to the same as Bambu Studio
- Wipe while retracting
- Wipe distance to 2mm and retract amount before wipe to 10%
- Turning reduce infill retraction off.

**Comment by user6508** [comment]
The same issue I have with orca and prusa.
Cura gives more clearer result.

**Comment by faberyx** [comment]
have the same issue too, tried a lot of different settings but I still get it, it doesn't happen at all on Cura

**Comment by Naninani** [comment]
I think the issue is with "Wipe while retracting" not working at all.

**"Wipe while retracting" disabled, "Retract amount before wipe" set to 100% or any other %**
<img width="1406" alt="image" src="https://github.com/SoftFever/OrcaSlicer/assets/1681582/63c2b38e-42c8-4092-a79e-a62ab8c87a06">
<img width="1254" alt="image" src="https://github.com/SoftFever/OrcaSlicer/assets/1681582/d403fd11-5ac7-4d38-822d-bedb6853ab04">

**"Wipe while retracting" enabled, "Retract amount before wipe" set to 50%**
<img width="1411" alt="image" src="https://github.com/SoftFever/OrcaSlicer/assets/1681582/c4c1a55c-9325-4424-af5c-5778bb326e5a">

**"Wipe while retracting" enabled, "Retract amount before wipe" set to 0%**
<img width="1314" alt="image" src="https://github.com/SoftFever/OrcaSlicer/assets/1681582/1eb9e615-f93e-4bc5-8444-bfbf75ffe36e">

@SoftFever, looks like `Wipe while retracting` is not working.

**Comment by yonkiman** [comment]
I'm seeing two problems related to this.  The first is that there is a lot of long (>100mm) travel without retracting.  So there's a lot of oozing out of my 0.6mm nozzle.

The second is that rotating the part in Z by 2 degrees (the only way I can put the seam close to where I want it ) radically changes the slicing, so there is a _ton_ more travel (again, with no retractions) than the identical but non-rotated part in the upper right.
![image](https://github.com/SoftFever/OrcaSlicer/assets/12636032/036bd375-3922-4b0b-a6ed-6278c5cb035e)
[Travel Changes.3mf.zip](https://github.com/SoftFever/OrcaSlicer/files/14215335/Travel.Changes.3mf.zip)

**Comment by Edzi11** [comment]
Ive been having fuzz and stringing as well that disappears when using prusa slicer. same settings as well.

**Comment by sixerstyle** [comment]
Same here, you can't get stringing under control.
no problem with cura or prusa slicer

**Comment by oscillatingdecay** [comment]
Wipe on retracting is not occurring for me.
![Screenshot 2024-04-05 at 6 42 52 PM](https://github.com/SoftFever/OrcaSlicer/assets/141880001/97813aff-9283-460a-887d-11e93fc008bc)

**Comment by grrminator** [comment]
Its very frustrating. I have multiple printers that were on cura.

As soon as I move to orca, which is much better, this stringing will NOT go away. I get that its fuzz and mostly can just be burned off, I shouldnt have to get a butane torch every time my printer finishes something.

**Comment by sixerstyle** [comment]
Since version 2.0.0 the stringing problems have disappeared for me

**Comment by staal54a** [comment]
> I think the issue is with "Wipe while retracting" not working at all.
> 
> **"Wipe while retracting" disabled, "Retract amount before wipe" set to 100% or any other %** <img alt="image" width="1406" src="https://private-user-images.githubusercontent.com/1681582/300076165-63c2b38e-42c8-4092-a79e-a62ab8c87a06.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjEyNzIxMDcsIm5iZiI6MTcyMTI3MTgwNywicGF0aCI6Ii8xNjgxNTgyLzMwMDA3NjE2NS02M2MyYjM4ZS00MmM4LTQwOTItYTc5ZS1hNjJhYjhjODdhMDYucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDcxOCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA3MThUMDMwMzI3WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NjhiOTIyOGYxNjg2OWEzOWYyNzI1ZTQ0NTM5ODZmODQ3MzMyYzU5ZmUwNTE0OWRmMTk4NzAyZjdjMDdjMWIxNSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.yqlFQaEKx1Sa9MlxHwQIWaQaegyN0PjywafiOARr5Lc"> <img alt="image" width="1254" src="https://private-user-images.githubusercontent.com/1681582/300077255-d403fd11-5ac7-4d38-822d-bedb6853ab04.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjEyNzIxMDcsIm5iZiI6MTcyMTI3MTgwNywicGF0aCI6Ii8xNjgxNTgyLzMwMDA3NzI1NS1kNDAzZmQxMS01YWM3LTRkMzgtODIyZC1iZWRiNjg1M2FiMDQucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDcxOCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA3MThUMDMwMzI3WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ODIyNGY1MGI5NjI5YWYyM2RhYjdlMzRmNTEzOWM5NGQwNmNjZmYyNzBhMTY2OGYwYWNjMTEwMWE0YjNmYTc4MiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.FyyKAqSl0yDVYh5F6NKC1IIDmadIpg98KNqLzU_W0I4">
> 
> **"Wipe while retracting" enabled, "Retract amount before wipe" set to 50%** <img alt="image" width="1411" src="https://private-user-images.githubusercontent.com/1681582/300076722-c4c1a55c-9325-4424-af5c-5778bb326e5a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjEyNzIxMDcsIm5iZiI6MTcyMTI3MTgwNywicGF0aCI6Ii8xNjgxNTgyLzMwMDA3NjcyMi1jNGMxYTU1Yy05MzI1LTQ0MjQtYWY1Yy01Nzc4YmIzMjZlNWEucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDcxOCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA3MThUMDMwMzI3WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9YWIwZmZlNTZkZGJjY2JhZWUwMzAxMTdlOTE3Y2Y2OGY5YmM4MjNjNWQ2MTQzYzdjZmI4NjA4NzI5ZmIxN2RjZSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.jptmoHu3sDN62Bw44eUvHnvewtFKZw6yc43PbMYIVfM">
> 
> **"Wipe while retracting" enabled, "Retract amount before wipe" set to 0%** <img alt="image" width="1314" src="https://private-user-images.githubusercontent.com/1681582/300077121-1eb9e615-f93e-4bc5-8444-bfbf75ffe36e.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjEyNzIxMDcsIm5iZiI6MTcyMTI3MTgwNywicGF0aCI6Ii8xNjgxNTgyLzMwMDA3NzEyMS0xZWI5ZTYxNS1mOTNlLTRiYzUtODQ0NC1iZmJmNzVmZmUzNmUucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDcxOCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA3MThUMDMwMzI3WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZjI4N2Y4MDljNTE1NzhmMTRjYTA1NjljNjk1MzhjYTlkNDk5NWJlZWQyOWZlOTliMDY5MDZlNThiYjcxZTI0YyZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.I6_pvV_sHrtnzq7HLIZvn0_owPmHa1EYxTqWzdblWc0">
> 
> Also for whatever reason the `Z hop type` on the inner wall after the wipe is `Normal` instead of `Spiral`. <img alt="image" width="1265" src="https://private-user-images.githubusercontent.com/1681582/300078783-58613fb1-4a68-47ca-8227-086b919b6111.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjEyNzIxMDcsIm5iZiI6MTcyMTI3MTgwNywicGF0aCI6Ii8xNjgxNTgyLzMwMDA3ODc4My01ODYxM2ZiMS00YTY4LTQ3Y2EtODIyNy0wODZiOTE5YjYxMTEucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDcxOCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA3MThUMDMwMzI3WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9YjY2N2MwNTk5ZTViZDc5NzlkOWUwNzFkMTAyNjMwNzc3MmJlYTM1YTIxYzQ5Y2Y0YjliMDVkMjM4ZWZjNzRkMSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.FYoD4GSLSFyxA7Cj09Rwf9-Umjpw-sISChumuKCWEvA">
> 
> @SoftFever, looks like `Wipe while retracting` is not working.

I can't seem to load some of your images but at least in that last image, if you zoom all the way in on where those retractions to z-hops are occurring you can see the yellow line indicating the wipe. It is a much smaller line than any of the walls (it's about the same size as the travel/z-hop lines versus teh printing wall lines)

**Comment by staal54a** [comment]
> Wipe on retracting is not occurring for me. ![Screenshot 2024-04-05 at 6 42 52 PM](https://private-user-images.githubusercontent.com/141880001/320166587-97813aff-9283-460a-887d-11e93fc008bc.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjEyNzIxMDcsIm5iZiI6MTcyMTI3MTgwNywicGF0aCI6Ii8xNDE4ODAwMDEvMzIwMTY2NTg3LTk3ODEzYWZmLTkyODMtNDYwYS04ODdkLTExZTkzZmMwMDhiYy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNzE4JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDcxOFQwMzAzMjdaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1hZDliZTM3Y2U3YzRlNmUzZGY1YTg2NDZmMzBlNDIxNDM1ZDcxYTk2OWIxNmEzMzBjMzVkYzg4NTRlNTAxN2IxJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.lc4L_HGlDx843QksaqOWdKC5pOfDSL_jxYEF44O2_Ww)

And similar for your issue, if you zoom in on the rightmost inner wall, there is a small yellow line above the inner wall line which indicates a wipe is occurring. The interesting part is why it's not occurring on all those retractions but also tough to gather without seeing the retraction settings and such.

I personally haven't had an issue getting wipe while retracting to work in any version of Orca I've used (since pre 1.0) but the main issue I'm seeing is that the retract amount before wipe percentage is not truly honored. Basically it seems that the wipe distance and speed is used to calculate the amount of retraction that occurs during the wipe, then any additional retraction required occurs before the wipe.

**Comment by ForsakenRei** [comment]
I'm also having stringing issue. Temp tower printed and chose a best temp for my filament(190C surprisingly) and printed a retraction test which is perfect, but when printing string tower or my normal parts I got tons of strings. When looking at it actually prints the strings tower, it seems my spiral z hop only happens on 1 of the 2 towers towards the tip so one of them will have a lot of strings. I don't know why it only do z hop on that one tower.

**Comment by DearTanker** [comment]
Because I use the Creality K1C, I have never tried Bambu Studio before.

But, recently I found that Bambu Studio can actually support non-Bambu 3D printers, so I tried it with the string hell model with it. I didn't expect that the result was very perfect. I have ever seen that good result before. After many days with countless comparisons and tweak with Orca, I finally discovered a detail that might be causing this problem, and you can't set it through Orca.

<img width="1917" alt="2024-09-15 162125" src="https://github.com/user-attachments/assets/28d9eafa-9faa-479e-9ba1-74a13fd00014">
[left: Bambu Studio][right: OrcaSlicer]

Bambu Studio will retraction again before traveling after retracting and wiping, while Orca will not do it no matter how I adjust the settings. I think this is the root cause of the strings problem.

**Comment by staal54a** [comment]
> Because I use the Creality K1C, I have never tried Bambu Studio before.
> 
> But, recently I found that Bambu Studio can actually support non-Bambu 3D printers, so I tried it with the string hell model with it. I didn't expect that the result was very perfect. I have ever seen that good result before. After many days with countless comparisons and tweak with Orca, I finally discovered a detail that might be causing this problem, and you can't set it through Orca.
> 
> <img alt="2024-09-15 162125" width="1917" src="https://private-user-images.githubusercontent.com/1122669/367569277-28d9eafa-9faa-479e-9ba1-74a13fd00014.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzAyNTE2ODUsIm5iZiI6MTczMDI1MTM4NSwicGF0aCI6Ii8xMTIyNjY5LzM2NzU2OTI3Ny0yOGQ5ZWFmYS05ZmFhLTQ3OWUtOWJhMS03NGExM2ZkMDAwMTQucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTAzMCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDEwMzBUMDEyMzA1WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NGUyMzQxMWRmMjQ4M2EwNDQyM2EzZWIwYjg4MzY3MzU1MTQxZTJhNjNkMzZiZDM2YWJjY2NhNjQ0N2M2OTg0NCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.FnZ8Nl8aPZ6hTWEQvPA8jRDcEuPSUQAooykjCeuvZmU"> [left: Bambu Studio][right: OrcaSlicer]
> Bambu Studio will retraction again before traveling after retracting and wiping, while Orca will not do it no matter how I adjust the settings. I think this is the root cause of the strings problem.

I'm pretty sure what you are seeing is the fact that Bambu Studio will respect the 0% retract before wipe value but Orca does not. So my guess as to what you are seeing is the following (which I was able to recreate in Bambu Studio and Orca Slicer):

- In Bambu Studio, with 0% retract before wipe, the wipe movement and initial retraction happen at the same time. If at the end of the wipe, the retraction length has not been reached, Bambu Studio inserts a retraction for the remaining amount. If you have a model with a bunch of retract and wipes on straight lines, the remaining retraction amount should be pretty consistent.
- In Orca Slicer, with 0% retract before wipe, the slicer will calculate the retraction length it would NOT be able to do within the requested wipe distance. It then attempts to retract that before it starts the wipe (unless the value is 0 in which case it skips this). So instead of a line finishing and the wipe starting immediately, there is really a (potential) pause while it does the initial retraction, then the wipe and retraction starts.

So I'm wondering if the wipe not starting immediately causes any issues since the nozzle would just be sitting there, albeit for a potentially really short period of time. I may play around with some different parameters (wipe speed and distance namely) to see if that improves the stringing issues I'm seeing myself.

**Comment by Milliw** [comment]
Just want to add my experience here, too.

Was using Cura since I got my first 3D printer. After gaining more knowledge I switched to OrcaSlicer. I find it far the better program. But stringing got more of an issue at sudden. Did excessive tests with a simple, small stringing tower stl. Tried different settings. Then printed same model with Cura and boom - stringing not an issue! So I switched back to Cura for now sadly.

But I do want to mention, that I tried PrucaSlicer as well and I got same results as with OrcaSlicer. Bummer.

Anycubic Mega S

Greetings

**Comment by MajorKingston** [comment]
Hey Guys! 
I had the same Problem on my Anycubic Kobra Plus with the stringing. The retraction test didnt have any differences whatsoever.

For me it was a Problem with the machine G-Code und the setting for Relative E.

Before the Fix, the Values for the extrusion in the G-Code were always positive float Numbers.
Afterwards they were negative an it was easy too see the steps in the retraction test G-Code.

I had to change this value in the machine G-Code, which sets back the extrusion value to 0 before the layer change, this value was commented out.

![Image](https://github.com/user-attachments/assets/92587fe8-0f3a-4f14-a977-cd52af48cb62)

Afterward i set my relative e  to on.

![Image](https://github.com/user-attachments/assets/696097db-bdb5-45ee-94e3-c3797d5d3901)

That fixed the stringing problem for me. Maybe its a solution for at least some of you.

**Comment by lapluviosilla** [comment]
Why was this closed? I have this issue too, and clearly other people too judging from Reddit.

**Comment by kadisonm** [comment]
It was closed automatically but yeah I switched completely back to Bambu Studio because of it.

**Comment by valerii-bokhan** [comment]
This issue can be resolved by reverting [this commit](https://github.com/valerii-bokhan/OrcaSlicer/commit/75328a2de86f77c7a06c02d3e3aba473d9980ba0).

@SoftFever FYI

- The last 5% of retraction length is being processed as a separate retraction command.
- It's still an actual code in Bambu Studio.
- Hard to say why it works better, but, as the original TODO comment says, `test and/or better math for this` is required.

Tests: Before and after reverting (Orca 2.3.2-dev, Wipe distance: 2mm, Retract before wipe: 80%)

<img width="1280" height="960" alt="Image" src="https://github.com/user-attachments/assets/394c1511-5063-43cd-8a4a-9e0b2ceb0c8e" />

<img width="1052" height="927" alt="Image" src="https://github.com/user-attachments/assets/aa8607e1-e2ce-4950-9ddf-fe4f39152531" />

**Comment by boromyr** [comment]
Were the objects from your tests printed as single objects that do not require nozzle travel moves in the air? It seems to reduce stringing thanks to the 5% retraction after the wipe and before the move, I think.

**Comment by valerii-bokhan** [comment]
Hm, no, these cylinders were used as "cooling towers", they were printed together with test duck models.

**Comment by boromyr** [comment]
Ok, as I was saying, I believe the improvement is due to the small retraction before the nozzle moves between the two objects; that is why the final 5% retraction leaves the seam cleaner. The subsequent G1 commands are movements without extrusion.

**Comment by valerii-bokhan** [comment]
However, it looks like a good thing, right?

It appears to be a hack in the code, but what would be the correct way to introduce/fix it?

The only idea I have is to add another setting - `Retract amount after wipe` or something like that. Because it's not 5% of the global retraction length, it's the last 5% of the retraction that occurred during the wipe.

**Comment by boromyr** [comment]
It seems like an interesting idea, especially after your comparison photos. Have you tried the retraction test model or multiple small cylinders to see what happens?
It would be interesting to try using high values for `Retract amount after wipe`, like 80% or 100%; it should emulate the coasting that many slicers use and see if it improves the seam by correctly calibrating it with `seam gap`.

**Comment by valerii-bokhan** [comment]
Right now, I am trying to use values like 80% before the wipe and 10% (of actual retraction length) after the wipe.

It gives me some promising results with a wipe distance = 0.4mm.

Before, I used a wipe distance of 2mm, and it mostly worked, but since I print very small models, I still had micro-stringing on the finer parts of the models. Now it looks better, but the strings are still there.

**My current settings:**

<img width="439" height="652" alt="Image" src="https://github.com/user-attachments/assets/65affc00-958e-4809-9997-bc84b0ae4075" />

**Photo of "cooling towers":**

- On the left - wipe distance 0.4mm, retraction after wipe - 10% of the retract length during wipe (hack in the code)
- On the right - wipe distance 0.4mm, but retraction after wipe - 10% of the global retraction length (made as a proper option in the settings)

<img width="1280" height="960" alt="Image" src="https://github.com/user-attachments/assets/cf70b818-a01b-41b3-bb19-e781dd038f34" />

**Sometimes seams are barely visible:**

<img width="1280" height="960" alt="Image" src="https://github.com/user-attachments/assets/a7b6dc7d-c176-4886-abce-52dda6221e31" />

I will definitely try out your idea of using high values (80-100%) of `Retract amount after wipe`.
I hope to post a draft PR during the weekend.

**Comment by boromyr** [comment]
And I will try it when you publish it, but the photos already speak for themselves. Specifically, my idea was that by setting it to 100% after purging, the retraction does not occur at the end of the wall, but the residual pressure in the nozzle still has the force to refill the seam gap during purging and close the wall gradually.

**Comment by valerii-bokhan** [comment]
> the photos already speak for themselves

Indeed, but the main models still have stringing around the seams, so I need to do more and more tests and post more photos of more complicated models than cylinders. :)

**Comment by valerii-bokhan** [comment]
- Retract before wipe: 80%
- After wipe: 20%
- No retract during wipe.
- Wipe distance: 0.4mm

Strings are still there - between tail and body + upper part of the beak.

<img width="1054" height="761" alt="Image" src="https://github.com/user-attachments/assets/a47a07f4-abbe-48c1-ad23-5ff2ffb771a2" />

<img width="960" height="1280" alt="Image" src="https://github.com/user-attachments/assets/01803471-edd2-4484-b3cd-3e409e51c161" />

<img width="1280" height="960" alt="Image" src="https://github.com/user-attachments/assets/ac20aa95-73d3-4c17-8bc8-18bda73ce433" />

<img width="1280" height="960" alt="Image" src="https://github.com/user-attachments/assets/3725f986-ff81-4f64-a3cb-691f2cd033ea" />

**Comment by valerii-bokhan** [comment]
- Retract before wipe: 20%
- After wipe: 80%
- No retract during wipe.
- Wipe distance: 0.2mm

In my case, it didn't work - overextrusion on the finer parts, probably because I use 1.1 as a `flow ratio` value for outer walls.

Sorry, it's probably hard to see in the photos... Seams are clean enough, but way more strings on the small/fine parts. 80/20 worked better.

Before/after comparison (the fresh result is on the right):
<img width="640" height="480" alt="Image" src="https://github.com/user-attachments/assets/f0b38333-d519-4d02-b214-6639e2faee94" />

<img width="640" height="480" alt="Image" src="https://github.com/user-attachments/assets/585e25aa-b083-46f0-80f2-db38196e4d6c" />

<img width="640" height="480" alt="Image" src="https://github.com/user-attachments/assets/498da403-ae41-498e-b571-3846288bfebb" />

<img width="640" height="480" alt="Image" src="https://github.com/user-attachments/assets/dcc902b7-904d-4180-9e7d-b3d98def93c1" />

In conclusion, more tests are needed, as always... PR will be posted tomorrow :)
